modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/PublicationsPortlet.java

```diff
@@ -99,6 +107,24 @@ public void render(
 		super.render(renderRequest, renderResponse);
 	}
 
+	@Activate
+	protected void activate() throws PortalException {
+		_companyLocalService.forEachCompanyId(
```

Comment:
@stsquared99 why does this need to run on every startup? The upgrade takes care of the preexist ones, and the new ones got taken care of during creation. This would just end up doing nothing, right?

Reply:
This would be in case of a new instance, if for whatever reason this portlet is loaded in after the company is initialized.
We talked about this, right @tinatian?

Reply:
@shuyangzhou Yes, this is needed to initialize resource permission for publication, same as what we did for portlet initialization.

Tina.